api/all/src/main/java/io/opentelemetry/api/trace/Span.java

```diff
@@ -162,6 +162,23 @@ default Span setAttribute(AttributeKey<Long> key, int value) {
     return setAttribute(key, (long) value);
   }
 
+  /**
+   * Sets attributes to the {@code Span}. If the {@code Span} previously contained a mapping for any
+   * of the keys, the old values are replaced by the specified values.
+   *
+   * @param attributes the attributes
+   * @return this.
+   */
+  @SuppressWarnings("unchecked")
+  default Span setAllAttributes(Attributes attributes) {
+    if (attributes == null || attributes.isEmpty()) {
+      return this;
+    }
+    attributes.forEach(
+        (attributeKey, value) -> this.setAttribute((AttributeKey<Object>) attributeKey, value));
```

Comment:
Thanks @HaloFour, this follows the pattern we have in a couple other places like `putAll` where we hide the unchecked cast from users with this sort of shortcut.